frontend/src/container/PopupSelectImage.jsx

```diff
@@ -0,0 +1,217 @@
+  constructor (props) {
+    super(props)
+    this.croppedImageRef = React.createRef()
+    this.state = {
+      imageBlobURL: null,
+      mode: MODE.SELECT,
+      errorMessage: null
+    }
+  }
+
+  componentWillUnmount () {
+    // INFO - SG - 20211109 - ensure the blob represented by imageBlobURL
+    // is properly de-allocated when the component is destroyed
+    const { state } = this
+    if (!state.imageBlobURL) return
+    console.log(`Revoking object URL ${state.imageBlobURL}`)
+    URL.revokeObjectURL(state.imageBlobURL)
+  }
+
+  handleDropFile = async (droppedFileList, rejectedFileList, event) => {
+    const { props } = this
+    if (!droppedFileList || droppedFileList.length === 0) return
+    const file = droppedFileList[0]
+
+    if (!ALLOWED_IMAGE_MIMETYPES.includes(file.type)) {
+      this.setState({ errorMessage: props.t('The type of this file is not allowed') })
+      return
+    }
+    if (props.maximumFileSize && file.size > props.maximumFileSize) {
+      this.setState({ errorMessage: props.t('The file is too big') })
+      return
+    }
+
+    this.setState({ imageBlobURL: URL.createObjectURL(file) })
+    // NOTE - 2021-11-10 - SG - Directly go to crop when the file has
+    // been added through a click + file selector.
+    // Going to crop in case of a drag event is handled by handleDragLeave()
+    // so that cypress upload() command properly works in functional tests.
+    if (!(event instanceof DragEvent)) this.setState({ mode: MODE.CROP })
+  }
+
+  handleDragLeave = () => {
+    this.setState({ mode: MODE.CROP })
+  }
+
+  handleValidate = async () => {
+    const { props } = this
+    const cropper = this.croppedImageRef.current.cropper
+    const blob = await new Promise((resolve) => {
+      cropper.getCroppedCanvas().toBlob(resolve)
+    })
+    props.onValidate(blob)
+  }
+
+  handleBack = async () => {
+    this.setState({ mode: MODE.SELECT })
+  }
+
+  render () {
+    const { props, state } = this
+    return (
+      <CardPopup
+        onClose={props.onClose}
+        onValidate={this.handleValidate}
+        label={state.mode === MODE.SELECT ? props.t('Select an image') : props.t('Crop the image')}
+        customColor={GLOBAL_primaryColor} // eslint-disable-line camelcase
+        faIcon='fas fa-upload'
+        customClass={classnames('PopupSelectImage', props.customClass)}
+      >
+        {state.mode === MODE.SELECT && (
+          <>
+            <Dropzone
+              onDrop={this.handleDropFile}
+              onDragLeave={this.handleDragLeave}
+              multiple={false}
+              className='PopupSelectImage__dropZone'
+            >
+              <SelectFilePlaceHolder t={props.t} />
+            </Dropzone>
+            {props.recommendedDimensions && (
+              <div>
+                <i className='fas fa-fw fa-expand-arrows-alt' /> {props.t('Recommended dimensions: {{dimensions}} px', { dimensions: props.recommendedDimensions })}<br />
```

Comment:
Do you need this `<br />`? It should be redundant with the div. If it's to have a blank line a margin may be better.

Reply:
A leftover, I added the `<div>` in this PR and did not remove `<br />`.